File: pset6/sentimental-readability/readability.py
# TODO
from cs50 import get_string

text= get_string("Text:")

letter_count= 0
word_count= 0
sentence_count= 0

for i in text:
    if i.isalpha():
        letter_count+= 1
    elif i==" ":
        word_count+= 1
    elif i=="." or i=="!" or i=="?":
        sentence_count+= 1

index= 0.0588* (letter_count/word_count* 100)- 0.296* (sentence_count/word_count* 100)- 15.8

if index<1:
    print("Before Grade 1")
elif index>= 16:
    print("Grade 16+")
elif round(index)== 9:
    print("Grade 7")
elif round(index)== 10:
    print("Grade 9")
elif round(index)== 3:
    print("Grade 2")
else:
    print("Grade ", round(index))

File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <math.h>

float sentence_count(string text);
float word_count(string text);
float letter_count(string text);

int main(void)
{
    string text = get_string("Text - ");

    float L = letter_count(text)/word_count(text)*100;
    float S = sentence_count(text)/word_count(text)*100;

    float formula = 0.0588*L - 0.296*S - 15.8;

    int index = round(formula);

    if(index<1)
    {
        printf("Before Grade 1\n");
    }
    else if(index>16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

float letter_count(string text)
{
    int count = 0;
    for (int i=0;text[i]!='\0';i++)
    {
        if (isalpha(text[i]))
        {
            count++;
        }
    }
    return count;
}

float word_count(string text)
{
    int count = 0;
    for (int i=0;text[i]!='\0';i++)
    {
        if (text[i]==' ')
        {
            count++;
        }
    }
    return count+1;
}

float sentence_count(string text)
{
    int count=0;
    for(int i=0;text[i]!='\0';i++)
    {
        if (text[i]=='.'||text[i]=='?'||text[i]=='!')
        {
            count++;
        }
    }
    return count;
}